src/server/scripts/EasternKingdoms/Karazhan/boss_shade_of_aran.cpp

```diff
@@ -503,6 +529,31 @@ class boss_shade_of_aran : public CreatureScript
                     return;
                 }
         }
+
+        void MoveInLineOfSight(Unit* who) override
+        {
+            ScriptedAI::MoveInLineOfSight(who);
+
+            if (SeenAtiesh || me->IsInCombat() || me->GetDistance2d(who) > me->GetAttackDistance(who) + 10.0f)
+                return;
+
+            Player* player = who->ToPlayer();
+            if (!player)
+                return;
+
+            for (auto id : AtieshStaves)
```

Comment:
The C++11 range-based for doesn't require auto, the Type name is also small in this case. Use the enum Type name or uint8 like @Tonghost said.